experimental/PropertyDDS/packages/property-binder/src/data_binder/dataBinder.ts

```diff
@@ -0,0 +1,2824 @@
+                                in_treeNode);
+                            fakeContext.getUserData().createdBindings = [];
+                            if (this._createBindingFromDefinition(fakeContext, definitions[0], rule.activationInfo)) {
+                                // A binding was created -- call back for postCreate
+                                io_instantiatedBindings.push(fakeContext as any);
+                            }
+                        } else {
+                            if (!(existingBinding instanceof definitions[0].bindingConstructor)) {
+                                console.warn(
+                                    'Specializing a DataBinding after DataBindings have already been instantiated for bindingType: ',
+                                    rule.bindingType
+                                );
+                            } else {
+                                (existingBinding as DataBinding)._incReferenceCount();
+                            }
+                        }
+                    }
+                }
+
+                // We recurse if another binding wants to recurse, or we want to recurse.
+                recurse = recurse || thisRecurse;
+            });
+
+            return recurse;
+        };
+
+        // Traverse the hierarchy and instantiate bindings
+        recursivelyVisitHierarchy(
+            in_rootPropertyElement, in_rootPropertyElement.getAbsolutePath(), this._dataBindingTree, generalVisitor
+        );
+    }
+
+    /**
+     * Recursively creates data bindings on existing properties for the provided registration handles. We create them
+     * in a depth-first fashion.
+     *
+     * @param in_activationRules - The array of rules to apply bindings for
+     *
+     * @private
+     * @hidden
+     */
+    private _createBindingsRetroactively(in_activationRules: ActivationType[]) {
+        const instantiated = [];
+
+        // Find all the path roots to start recursing from. We end with all the deepest starting
+        // points for the set of handles.
+        const pathRoots = minimalRootPaths(_.map(in_activationRules, 'startPath'));
+
+        // Batch the rules based on these roots. Some of these rules may be a simple traversal to do, while
+        // others may require careful checking of references etc.
+        const rootsToRules = {};
+        for (let i = 0; i < pathRoots.length; ++i) {
+            const rules: any[] = [];
+            for (let j = 0; j < in_activationRules.length; j++) {
+                if (in_activationRules[j].startPath.indexOf(pathRoots[i]) === 0) {
+                    // This rule is encompassed in this subtree
+                    rules.push(in_activationRules[j]);
+                }
+            }
+            rootsToRules[pathRoots[i]] = rules;
+        }
+
+        // For each root, recursively traverse the property hierarchy and instantiate bindings.
+        _.each(rootsToRules, (in_rules: any, in_root: any) => {
```

Comment:
`(in_rules: any, in_root: any)` could be `(in_rules: ActivationType[], in_root: string)`